src/scaffold/ScaffoldView.tsx

```diff
@@ -123,71 +107,99 @@ const FirstPage: React.FC<PageProps> = ({ onNext }) => {
 
 const SecondPage: React.FC<PageProps> = ({ onPrevious }) => {
   const { state, actions } = useStore();
-  const [selectedBundler, setSelectedBundler] = React.useState<string | undefined>(undefined);
+  const options = state.options;
+
+  const { repoType, template, name, bundler, targetFolder } = options;
+  const canScaffold = repoType !== '' && template !== '' && name !== '' && bundler !== '' && targetFolder !== '';
+
+  const openLocalPathModal = () => {
+    actions.selectLocalPath();
+  };
+
+  const createPiralPilet = () => {
+    actions.scaffold(options);
```

Comment:
Why? The options are anyway placed in the global store now - so just `actions.scaffold()` and get the options from the store inside.

Reply:
I always get an empty initial state if I get options from the store.

![Screenshot 2022-04-28 171723](https://user-images.githubusercontent.com/55950527/165786275-34eceba8-298f-407c-8abd-ea1b06e81add.png)

Reply:
Because you take the *initial* state, not the *current* state. Just take the current state - and you are good.

For this you can either use `useStore.getState()` or introduce the second argument (besides "set" there is a "get") or abuse the `set` as we did on Tuesday. Nothing new - all things that you did / have seen already.